src/classes/Accounts.cls

```diff
@@ -0,0 +1,194 @@
+/*
```

Comment:
Though I like the domain class name representing the object, I'm a bit concerned that this can get confusing in usage. We often represent "Accounts" and "Contacts" as collection vars throughout the code. For all of the domain classes, maybe names like one of the following:

- AccountDomain, ContactDomain, HouseholdDomain
- DomainAccount, DomainContact, DomainHousehold
- DOM_Account, DOM_Contact, DOM_Household
- others?

Reply:
This is a good callout, and highlights some of the benefits of adopting a domain-driven model.  Working with a collection of Account records is a form of primitive-obsession.  The idea is that instead of using raw collections of Account records directly (in Lists and Maps) the code should be using instances of Domain classes when working with collections of these records.  Every time you have a collection of Account Ids, you instantiate Accounts and work with the object rather than the collection of underlying records directly.  

Once NPSP has fully adopted domain driven design, one will use the Factory pattern to get an instance of the Domain and work with that - `Application.Domain.getInstance(listAccountIds)`

I suggest keeping the class names as Accounts, Contacts, etc. and continuing to push in the direction of eliminating the instances in the code that store and interact with low-level collections of SObject records.